src/components/ProductListing/product-card/AddToWishlistBtn.jsx

```diff
@@ -1,12 +1,19 @@
 import {Link} from "react-router-dom";
-const AddToWishlistBtn = () => 
-    <button className="outline-btn p-0 b-radius-2 text-bold card-wishlist m-5 flex-row justify-content-center align-center flex-gap-1">
-        <span className="wishlist-icon">
-            <i className="far fa-heart"></i>
-        </span>
-        <p className="wishlist-text">Add to wishlist</p>
-    </button>
+import { addToWishlistHandler, removeFromWishlistHandler } from "../../../utils" 
+import { useWishlist } from "../../../context";
 
+const AddToWishlistBtn = (props) => {
+    const { wishlistDispatch } = useWishlist();
+    return (
+        <button onClick={(e) => addToWishlistHandler(e, props.productData, props.token,wishlistDispatch)} className="outline-btn p-0 b-radius-2 text-bold card-wishlist m-5 flex-row justify-content-center align-center flex-gap-1">
```

Comment:
although it is not necessary but you can destructure producData, token from props object for better readability.